aten/src/ATen/Context.h

```diff
@@ -307,9 +311,15 @@ class TORCH_API Context {
       lazyInitHIP();
     }
   }
+  void initXPUIfNeeded(c10::DeviceType p) {
+    if (p == c10::DeviceType::XPU) {
+      lazyInitXPU();
+    }
+  }
```

Comment:
why these are part of allocator PR?

Reply:
Here calls `getXPUHooks().initXPU()` to initialize device caching allocator per device.